### PR TITLE
CDPT-1511 Intranet migration add microsoft login azure ad

### PR DESCRIPTION
**Pull request by EarthlingDavey**
Continues on from work in https://github.com/ministryofjustice/intranet/pull/507

Adds support for login via Azure AD/Entra OAuth.

@wilson1000 I understand that we probably don't want to keep the Azure Oauth on dev for long, but can we add it in the short term for testing?

### CODE REVIEW

**Comment by wilson1000** [comment]
Hey mate, this looks good!
I've reviewed some but will leave the rest until the morning as it looks like there are a couple of things left on it but, really nice to see 😄

**Comment by EarthlingDavey** [comment]
@wilson1000 is it ok with you if I spend an hour or 2 improving the code for this. I think it's so important that this is easy to read & maintain! `#nopressure`

If you've written some review comments already, would you mind submitting them please? Or, if you're mid-review, I can hold off.

Cheers :)

**Review by wilson1000 — COMMENTED** [review]
There are some conflicts but in general, this looks good.
I'd be really interested @EarthlingDavey to join a call with you just before this is launched on to development

File: deploy/development/config.yml [hunk 1]
```diff
@@ -7,3 +7,13 @@ data:
   WP_ENV: "development"
   WP_HOME: 'https://dev-intranet.apps.live.cloud-platform.service.justice.gov.uk'
   WP_SITEURL: 'https://dev-intranet.apps.live.cloud-platform.service.justice.gov.uk/wp'
+  # See Azure Setup for more information on how to get these values.
+  OAUTH_CLIENT_ID: 1dac3cbf-91d2-4c0e-9c80-0bf3f8fabd75
+  OAUTH_TENNANT_ID: c6874728-71e6-41fe-a9e1-2e8c36776ad8
```
[the remark this answers is not in the document]
Comment (EarthlingDavey) [reply deploy/development/config.yml:12]: I think it's ok for them to be public, as long as `OAUTH_CLIENT_SECRET` is in a secret.

Other projects have them public - and having them in config might make things easier for us to manage.

Based on our recent chat, I'll deploy now, happy to change at a later date.

File: .github/README.md [hunk 2]
```diff
@@ -265,6 +265,49 @@ To verify that S3 & CloudFront are working correctly.
...
+    - Admin consent description: `Access Intranet on behalf of the signed-in user`
+    - User consent display name: `Access Intranet`
+    - User consent description: `Access Intranet on your behalf`
+14. Click on `Add a client application`.
+15. Enter the Client ID of the application you created.
+16. Check the box next to the application you created.
+17. Click on `Add application`.
+
+The oauth2 flow should now work with the Azure AD/Entra ID application.
+You can get an Access Token, Refresh Token and an expiry of the token.
```
[the remark this answers is not in the document]
Comment (EarthlingDavey) [reply .github/README.md:311]: Sounds good!

**Pull request merged**